include/boost/random/hyperexponential_distribution.hpp

```diff
@@ -0,0 +1,797 @@
+
+            assert( hyperexp_detail::check_params(probs_, rates_) );
+        }
+
+        /**
+         * Constructs a \c param_type from the <em>rate vector</em> parameter
+         * of the distribution and with equal phase probabilities.
+         *
+         * The <em>rate vector</em> parameter is given by the
+         * <a href="http://en.cppreference.com/w/cpp/language/list_initialization">brace-init-list</a>
+         * defined by \a l1, and the <em>phase probability vector</em> parameter is
+         * set to the equal phase probabilities (i.e., to a vector of the same
+         * length \f$k\f$ of the <em>rate vector</em> and with each element set
+         * to \f$1.0/k\f$).
+         *
+         * \param l1 The initializer list for inizializing the rate vector.
+         */
+        public: param_type(std::initializer_list<RealT> l1)
+        : probs_(std::distance(l1.begin(), l1.end()), 1), // Will be normalized below
+          rates_(l1.begin(), l1.end())
+        {
+            hyperexp_detail::normalize(probs_);
+
+            assert( hyperexp_detail::check_params(probs_, rates_) );
+        }
+#endif // BOOST_NO_CXX11_HDR_INITIALIZER_LIST
+
+        /**
+         * Gets the <em>phase probability vector</em> parameter of the distribtuion.
+         *
+         * \return The <em>phase probability vector</em> parameter of the distribution.
+         *
+         * \note
+         *  The returned probabilities are the normalized version of the ones
+         *  passed at construction time.
+         */
+        public: std::vector<RealT> probabilities() const
+        {
+            return probs_;
+        }
+
+        /**
+         * Gets the <em>rate vector</em> parameter of the distribtuion.
+         *
+         * \return The <em>rate vector</em> parameter of the distribution.
+         */
+        public: std::vector<RealT> rates() const
+        {
+            return rates_;
+        }
+
+        /** Writes a \c param_type to a \c std::ostream. */
+        public: BOOST_RANDOM_DETAIL_OSTREAM_OPERATOR(os, param_type, parm)
+        {
+            detail::print_vector(os, parm.probs_);
+            os << ' ';
+            detail::print_vector(os, parm.rates_);
+
+            return os;
+        }
+
+        /** Reads a \c param_type from a \c std::istream. */
+        public: BOOST_RANDOM_DETAIL_ISTREAM_OPERATOR(is, param_type, parm)
+        {
+            std::vector<RealT> tmp;
+
+            detail::read_vector(is, tmp);
+            if (!is)
+            {
+                return is;
+            }
+            parm.probs_.swap(tmp);
+            hyperexp_detail::normalize(parm.probs_);
+
+            is >> std::ws;
+
+            tmp.clear();
+            detail::read_vector(is, tmp);
+            if (!is)
+            {
+                return is;
+            }
+            parm.rates_.swap(tmp);
+
+            //pre: vector size conformance
+            assert(parm.probs_.size() == parm.rates_.size());
```

Comment:
assert is for programming errors.  assert should never be used to validate input.